### PR TITLE
refactor: deprecate `Reply#getResponseTime()` in favour of `Reply#elapsedTime`

### LINKED ISSUE #4575: Proposed updates to the reply.getResponseTime() documentation
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

The `.getResponseTime()` portion of the docs states `Note that unless this function is called in the [onResponse hook](https://www.fastify.io/docs/latest/Reference/Hooks#onresponse) it will always return 0.`. However, I've tested it with `onResponse` and it doesn't return 0, it returns a very small number. This behaviour is expected, but is not represented in the docs.

I'd also like to add something to say that if you'd like to record route response time metrics, this is the place to do it.

What are your thoughts?

Reference: https://www.fastify.io/docs/latest/Reference/Reply/#getresponsetime

### Motivation

Improve documentation & show how easy it is to response response time reliably (much harder in other frameworks)

### Example

Note: To measure the ensure request response time, the function must be called in a `onResponse` hook. Calling the function in other hooks will result in `0` or very low numbers that do not represent the entire request response time.

If you would like to record request response time metrics, the result of `.getResponseTime()` in a `onResponse` hook is the recommended way to do that.

### PR DESCRIPTION
**Pull request by codershiba**
#### Checklist
Fixes #4575 

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/reply.js [hunk 1]
```diff
@@ -84,6 +84,17 @@ Object.defineProperties(Reply.prototype, {
...
+    get () {
+      let elapsedTime = 0
+
+      if (this[kReplyStartTime] !== undefined) {
+        elapsedTime = (this[kReplyEndTime] || now()) - this[kReplyStartTime]
+      }
+
+      return elapsedTime
+    }
+  },
```
Comment (mcollina) [inline lib/reply.js:97]: Can you please use this property in `getResponseTime`?

File: lib/warnings.js [hunk 2]
```diff
@@ -74,7 +74,12 @@ const FSTDEP018 = createDeprecation({
 
 const FSTDEP019 = createDeprecation({
   code: 'FSTDEP019',
-  message: 'reply.context property access is deprecated. Please use "request.routeOptions.config" or "request.routeOptions.schema" instead for accessing Route settings. The "reply.context" will be removed in `fastify@5`.'
+  message: 'You are accessing the deprecated "reply.context" property. Use "request.routeOptions.config" or "request.routeOptions.schema" instead for accessing Route settings. Property "reply.context" will be removed in `fastify@5`.'
```
Comment (mcollina) [inline lib/warnings.js:77]: why this change?

File: lib/reply.js [hunk 3]
```diff
@@ -452,7 +463,10 @@ Reply.prototype.callNotFound = function () {
   return this
 }
 
+// TODO: should be removed in fastify@5
 Reply.prototype.getResponseTime = function () {
+  FSTDEP020()
```
Comment (mcollina) [inline lib/reply.js:468]: Can you add a test to check this is correctly emitted. You can modify one of the existing tests for `getResponseTime()`.

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by codershiba** [comment]
@mcollina Made the requested changes, let me know if I need to do anything else.

File: test/internals/reply.test.js [hunk 4]
```diff
@@ -1640,6 +1654,67 @@ test('reply.getResponseTime() should return the same value after a request is fi
...
+  })
+
+  fastify.addHook('preValidation', (req, reply, done) => {
+    t.not(reply.elapsedTime, reply.elapsedTime)
+    done()
+  })
+
+  fastify.addHook('onResponse', (req, reply) => {
+    t.end()
+  })
```
Comment (metcoder95) [inline test/internals/reply.test.js:1694]: Maybe refactor this test to compare the `elapsedTime` within `preValidation` and later within `onResponse` to verify the latter is greater than the former?

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you please update

https://github.com/fastify/fastify/blob/0be27ca0cedf6e74a499f18e73d55aa0ef353b21/lib/reply.js#L816

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
seems the tests are failing on CI

**Comment by codershiba** [comment]
> seems the tests are failing on CI

Fixed

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/reply.js [hunk 5]
```diff
@@ -84,6 +84,17 @@ Object.defineProperties(Reply.prototype, {
...
+  elapsedTime: {
+    get () {
+      let elapsedTime = 0
+
+      if (this[kReplyStartTime] !== undefined) {
+        elapsedTime = (this[kReplyEndTime] || now()) - this[kReplyStartTime]
+      }
+
+      return elapsedTime
+    }
```
Comment (jsumners) [inline lib/reply.js:96]:
```suggestion
    get () {
      if (this[kReplyStartTime] === undefined) {
        return 0
      }
      return (this[kReplyEndTime] || now()) - this[kReplyStartTime]
    }
```

**Review by jsumners — APPROVED** [review]

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**